.github/workflows/contracts-coverage.yml

```diff
@@ -0,0 +1,43 @@
+name: Contracts-Coverage
+
+on:
+  push: 
+    branches: [ main ]
+
+env:
+  cwd: ${{github.workspace}}/packages/contracts
+
+concurrency: 
+  group: ${{ github.workflow }}-${{ github.head_ref || github.run_id }}
+  cancel-in-progress: true
+
+jobs:
+
+   coverage:
+    runs-on: ubuntu-20.04
+
+    steps:
+      - uses: actions/checkout@v3
+
+      - name: Use Node.js 16
+        uses: actions/setup-node@v2
+        with:
+          node-version: 16
+
+      - name: Install dependencies
+        run: yarn install --frozen-lockfile
+          
+      - name: Lint
+        run: |
+          yarn workspace @quadratic-funding/contracts run lint:solidity
+          yarn workspace @quadratic-funding/contracts run lint:js
+
+      - name: Test Contract
+        run: |
+          yarn workspace @quadratic-funding/contracts run generate:abi
+          yarn workspace @quadratic-funding/contracts run test:coverage
+
+      - name: Upload coverage report
+        uses: codecov/codecov-action@v2
```

Comment:
does this cost $$?

Reply:
It's entirely free for open source project.